.github/workflows/concretize-template.yaml

```diff
@@ -13,19 +13,19 @@ jobs:
       fail-fast: false
     steps:
       - name: Checkout Spack
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           repository: spack/spack
```

Comment:
Do we want to checkout the same version of spack and key4hep-spack as specified inside `.latest-commit` and `.key4hep-commit` for consistency? My experience with spack is that the latest versions are typically broken.